tests/cpp/test_segmentation.cpp

```diff
@@ -750,19 +750,31 @@ TEST_F(NVFuserTest, RevertPrivatizedUpcast) {
   fusion.addInput(tv0);
 
   auto tv1 = segment_set(tv0);
-  auto tv2 = castOp(DataType::Float, tv1);
 
-  auto tv3 = sum(tv2, {1});
-  fusion.addOutput(tv3);
+  auto tv2 = set(tv1);
```

Comment:
Slightly changed to trigger the assertion of `stablyOrderedExprs()`. Without the fix, this test results in:

```
[ RUN      ] NVFuserTest.RevertPrivatizedUpcast
unknown file: Failure
C++ exception with description " INTERNAL ASSERT FAILED at "/home/nmaruyama/scratch/nvfuser/debug7/csrc/fusion_segmenter.cpp":2619, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Expected ordered_exprs.size()
 and exprs().size() to be equal, but found 4 vs 5. exprs() doesn't form a DAG.
Exception raised from stablyOrderedExprs at /home/nmaruyama/scratch/nvfuser/debug7/csrc/fusion_segmenter.cpp:2619 (most recent call first):
```